src/wp-includes/rest-api/endpoints/class-wp-rest-attachments-controller.php

```diff
@@ -1444,4 +1454,42 @@ protected function get_edit_media_item_args() {
 		);
 	}
 
+	/**
+	 * Issue an incorrect-usage error when a featured image has been assigned to an attachment with
+	 * a MIME type that doesn't support post thumbnails.
+	 *
+	 * @see https://core.trac.wordpress.org/ticket/41692
+	 *
+	 * @param int    $attachment_id Attachment ID.
+	 * @param string $method        Method name.
+	 */
+	private function maybe_did_featured_media_wrong( $attachment_id, $method ) {
```

Comment:
Instead of this method, let's extend `handle_featured_media`. Something like this. 

```
protected function handle_featured_media( $featured_media, $post_id ) {
    if ( wp_attachment_is( 'audio', $post_id ) && post_type_supports( 'attachment:audio', 'thumbnail' ) || current_theme_supports( 'post-thumbnails', 'attachment:audio' ) ) {
         return parent::handle_featured_media( $featured_media, $post_id );
    } else if ( wp_attachment_is( 'video', $post_id ) && post_type_supports( 'attachment:video', 'thumbnail' ) || current_theme_supports( 'post-thumbnails', 'attachment:video' ) ) {
         return parent::handle_featured_media( $featured_media, $post_id );
    } else if(  wp_attachment_is( 'image', $post_id ) && current_theme_supports( 'post-thumbnails', $this->post_type ) && post_type_supports( $this->post_type, 'thumbnail' ) ){
          return parent::handle_featured_media( $featured_media, $post_id );
  }
   return new WP_Error( 'rest_feature_meta_update',
					sprintf(
					/* translators: %s: attachment mime type */
					__( 'This site does not support post thumbnails on attachments with MIME type %s.' ),
					$post->post_mime_type
				),
					array( 'status' => rest_authorization_required_code() )
		);
}
```

Reply:
I'm concerned that returning an error in this case will lead to a poor developer experience for a few reasons.

1. Unless more code is added to roll back the work done by the method up to this point, the attachment will be left in an partially created or partially updated state in the database. 
2. An error will be returned to the client, but, again, the attachment will have been partially created or updated. For a create operation, if the client tries to correct the request and send it again, a second attachment will be created.
3. The punishment of an error seems to me to not really fit the crime. Is setting the featured image on the wrong MIME type that big a deal?
4. There is some precedent for a `_doing_it_wrong()`, for example https://github.com/WordPress/wordpress-develop/blob/b17064a082b82fc737b312d9fb2849f2cccf792e/src/wp-includes/rest-api/endpoints/class-wp-rest-posts-controller.php#L2560-L2573.

For these reasons, if `_doing_it_wrong()` isn't going to work in this case, I think my next move would be to skip the error altogether rather than halt processing.

What do you think?